game/final_obs_gen.py

```diff
@@ -0,0 +1,494 @@
+                    [1, 1, 0, 1, 1, 0, 1, 1],
+                    [1, 1, 0, 1, 1, 0, 1, 1],
+                    [1, 1, 0, 1, 1, 0, 1, 1],
+                    [1, 1, 1, 1, 1, 1, 1, 1],
+                    [1, 1, 1, 1, 1, 1, 1, 1],
+                    [0, 1, 1, 1, 1, 0, 0, 0],
+                    [0, 0, 0, 1, 1, 0, 0, 0],
+                    [0, 0, 0, 1, 1, 0, 0, 0],
+                ]
+            )
+        elif self.obs_type == "C":
+            self.ARRAY = np.array(
+                [
+                    [0, 0, 0, 1, 1, 0, 0, 0],
+                    [0, 0, 0, 1, 1, 0, 0, 0],
+                    [0, 0, 0, 1, 1, 0, 1, 1],
+                    [0, 0, 0, 1, 1, 0, 1, 1],
+                    [0, 0, 0, 1, 1, 0, 1, 1],
+                    [0, 0, 0, 1, 1, 0, 1, 1],
+                    [1, 1, 0, 1, 1, 1, 1, 1],
+                    [1, 1, 0, 1, 1, 1, 1, 1],
+                    [1, 1, 0, 1, 1, 1, 0, 0],
+                    [1, 1, 1, 1, 1, 1, 0, 0],
+                    [1, 1, 1, 1, 1, 1, 0, 0],
+                    [0, 1, 1, 1, 1, 1, 0, 0],
+                    [0, 0, 0, 1, 1, 1, 0, 0],
+                    [0, 0, 0, 1, 1, 1, 0, 0],
+                ]
+            )
+        elif self.obs_type == "b":
+            self.ARRAY = np.array(
+                [
+                    [0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0],
+                    [0, 0, 0, 1, 1, 0, 1, 1, 0, 0, 0, 0, 0, 0],
+                    [0, 0, 1, 1, 1, 0, 0, 1, 1, 0, 0, 0, 0, 0],
+                    [0, 1, 1, 1, 1, 1, 0, 1, 1, 1, 0, 0, 0, 0],
+                    [1, 1, 1, 1, 1, 1, 0, 1, 1, 1, 0, 0, 0, 0],
+                    [0, 0, 0, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1],
+                    [0, 0, 0, 0, 0, 1, 1, 1, 1, 1, 1, 1, 0, 0],
+                    [0, 0, 0, 0, 0, 0, 1, 1, 1, 1, 1, 1, 1, 0],
+                    [0, 0, 0, 0, 0, 0, 0, 1, 1, 1, 1, 0, 0, 0],
+                ]
+            )
+        else:
+            self.ARRAY = np.array(
+                [
+                    [0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0],
+                    [0, 0, 0, 1, 1, 0, 1, 1, 0, 0, 0, 0, 0, 0],
+                    [0, 0, 1, 1, 1, 0, 0, 1, 1, 0, 0, 0, 0, 0],
+                    [0, 1, 1, 1, 1, 1, 0, 1, 1, 1, 0, 0, 0, 0],
+                    [1, 1, 1, 1, 1, 1, 0, 1, 1, 1, 0, 0, 0, 0],
+                    [0, 0, 0, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1],
+                    [0, 0, 0, 0, 0, 1, 1, 1, 1, 1, 1, 1, 0, 0],
+                    [0, 0, 0, 0, 0, 0, 1, 1, 1, 1, 1, 1, 1, 0],
+                    [0, 0, 0, 0, 0, 0, 0, 1, 1, 1, 1, 0, 0, 0],
+                ]
+            )
+
+    def setup_parameters(self):
+        self.height, self.width = self.ARRAY.shape
+        if self.obs_type in ("c", "C"):
+            self.y = self.REF_AXIS - self.height * self.GRID_SIZE
+        elif self.obs_type == "b":
+            self.y = self.LOW_BIRD_AXIS - self.height * self.GRID_SIZE
+        else:
+            self.y = self.HIGH_BIRD_AXIS - self.height * self.GRID_SIZE

```

Comment:
As mentioned, all this use of `GRID_SIZE` can potentially be eliminated.

Reply:
Eliminated now